stripe.go

```diff
@@ -394,38 +371,29 @@ func (s *BackendImplementation) Do(req *http.Request, body *bytes.Buffer, v inte
 		}
 
 		if err != nil {
-			if s.LogLevel > 0 {
-				s.Logger.Printf("Request failed with error: %v\n", err)
-			}
+			s.LeveledLogger.Errorf("Request failed with error: %v", err)
 		} else {
 			resBody, err := ioutil.ReadAll(res.Body)
 			res.Body.Close()
 			if err != nil {
-				if s.LogLevel > 0 {
-					s.Logger.Printf("Cannot read response: %v\n", err)
-				}
+				s.LeveledLogger.Errorf("Cannot read response: %v", err)
 			} else {
-				if s.LogLevel > 0 {
-					s.Logger.Printf("Request failed with body: %s (status: %v)\n", string(resBody), res.StatusCode)
-				}
+				s.LeveledLogger.Errorf("Request failed with body: %s (status: %v)",
+					string(resBody), res.StatusCode)
 			}
 		}
 
 		sleepDuration := s.sleepTime(retry)
 		retry++
 
-		if s.LogLevel > 1 {
-			s.Logger.Printf("Initiating retry %v for request %v %v%v after sleeping %v\n",
-				retry, req.Method, req.URL.Host, req.URL.Path, sleepDuration)
-		}
+		s.LeveledLogger.Infof("Initiating retry %v for request %v %v%v after sleeping %v",
```

Comment:
should this be Warnf?

Reply:
Hmm, thinking about it more deeply, I could see it either way. It is probably not actually a concerning operation, so `Infof` might be fine, but it's also a little unusual, so I could see `Warnf` too.

I'll change to `Warnf` just to make it more distinctive from other `Infof` messages.

Reply:
I was mostly going from the fact that it used to be `if s.LogLevel > 1 {`

Reply:
Oh I see. Yeah, I think I messed up the port.